Bottom: a360c273c25b01ea6af87349643fc36644dc4f25
Top:    f46231530a1fd948e0a0887741c438107f50f468
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-27 17:36:09 +0100

Refresh of hid-gtk-ask-for-polygons-after

---

diff --git a/src/draw.c b/src/draw.c
index a7d9832..e5a4d14 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -765,9 +765,11 @@ DrawMask (BoxType * screen)
   if (thin)
     gui->set_color (Output.pmGC, PCB->MaskColor);
   else
-    DrawMaskBoardArea (HID_MASK_BEFORE, screen);
+    {
+      DrawMaskBoardArea (HID_MASK_BEFORE, screen);
+      gui->use_mask (HID_MASK_CLEAR);
+    }
 
-  gui->use_mask (HID_MASK_CLEAR);
   r_search (PCB->Data->pin_tree, screen, NULL, clearPin_callback, &info);
   r_search (PCB->Data->via_tree, screen, NULL, clearPin_callback, &info);
   r_search (PCB->Data->pad_tree, screen, NULL, clearPad_callback, &info);
@@ -775,9 +777,10 @@ DrawMask (BoxType * screen)
   if (thin)
     gui->set_color (Output.pmGC, "erase");
   else
-    DrawMaskBoardArea (HID_MASK_AFTER, screen);
-
-  gui->use_mask (HID_MASK_OFF);
+    {
+      DrawMaskBoardArea (HID_MASK_AFTER, screen);
+      gui->use_mask (HID_MASK_OFF);
+    }
 }
 
 static void
